pages/components/docs/core-concepts

```diff
@@ -0,0 +1,71 @@
+# Core Concepts
+
+## The `as` prop
+The `as` prop is a feature that all of our components get from [styled-components](https://www.styled-components.com). It allows you to pass a HTML tag or another component to a Primer Component to be rendered as the base tag of that component.
+
+
+For example, say you are using a `Button` component, and you really need to apply `Flex` styles to it. You can compose `Flex` and `Button` like so:
+
+```
+<Button as={Flex}/>
+```
+
+This will allow you to use all of the `Button` props _and_ all of the `Flex` props without having to wrap your `Button` component in another `Flex` component.
+
+This pattern does have some limitations. Say you are using a `<Button>` component and you really want it to use an `a` tag, but you also need flex props. In this case, you'd need to wrap your `Button` component in a `Flex` and use the `as` prop on `Button` to set it to an `a` tag. You cannot provide multiple values for the `as` prop.
+
+
+```
+<Flex>
+  <Button as='a' />
+</Flex>
+```
+
+One alternate way to do this is to define `Flex` using the `as` prop set to `a` and then pass the custom `Flex` to the `Button`'s `as` prop, but that can feel a bit messy:
+
+```
+const LinkFlex = <Flex as='a'/>
+//in render function...
+<Button as={LinkFlex}/>
+
+```
+
+
+## Responsive props
+It's really easy to set different values for most of our component props based on screen size! We take advantage of [styled-system](https://github.com/styled-system/styled-system)'s responsive props API in our components.
```

Comment:
> most of our component props

Aren't _all_ system props responsive?

Reply:
All system props are, not all component props

Reply:
Should we clarify that all systems props are responsive here?

Also, are any non-system props responsive?

Reply:
No non-system props are responsive

Reply:
There isn't really an easy way to let a user know which props are "system props" and which aren't - that's why I intentionally left this vague.